### PR TITLE
 Continue ongoing simplification of sql_schema_differ 

### PR DESCRIPTION
**Pull request by tomhoule**
Just for peace of mind before I focus more on product issues.

### CODE REVIEW

File: migration-engine/connectors/sql-migration-connector/src/pair.rs [hunk 1]
```diff
@@ -3,10 +3,10 @@ use sql_schema_describer::{
     ColumnId, SqlSchema, TableId,
 };
 
-#[derive(Debug, Default, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+#[derive(Debug, Default, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
 pub(crate) struct Pair<T> {
-    previous: T,
-    next: T,
+    pub(crate) previous: T,
+    pub(crate) next: T,
```
Comment (tomhoule) [inline migration-engine/connectors/sql-migration-connector/src/pair.rs:9]: This should even be `pub`. There is no privacy to have here, it's a pair of things, it should only have exactly these two fields.

File: migration-engine/connectors/sql-migration-connector/src/sql_schema_differ/column.rs [hunk 2]
```diff
@@ -3,112 +3,91 @@ use enumflags2::BitFlags;
...
-        if self.previous.name() != self.next.name() {
-            changes |= ColumnChange::Renaming;
-        };
-
-        if self.arity_changed() {
-            changes |= ColumnChange::Arity
-        };
+pub(crate) fn all_changes(cols: Pair<ColumnWalker<'_>>, flavour: &dyn SqlFlavour) -> ColumnChanges {
+    let mut changes = BitFlags::empty();
+    let type_change = flavour.column_type_change(cols);
```
Comment (tomhoule) [inline migration-engine/connectors/sql-migration-connector/src/sql_schema_differ/column.rs:8]: Same code, one level of indentation lower.

File: migration-engine/connectors/sql-migration-connector/src/sql_schema_differ.rs [hunk 3]
```diff
@@ -462,73 +433,28 @@ impl<'schema> SqlSchemaDiffer<'schema> {
...
-        }
-
-        alter_enum.previous_usages_as_default = previous_usages_as_default;
-    }
-}
-
 /// Compare two [ForeignKey](/sql-schema-describer/struct.ForeignKey.html)s and return whether they
 /// should be considered equivalent for schema diffing purposes.
-fn foreign_keys_match(fks: Pair<&ForeignKeyWalker<'_>>, flavour: &dyn SqlFlavour) -> bool {
+fn foreign_keys_match(fks: Pair<&ForeignKeyWalker<'_>>, flavour: &dyn SqlFlavour, db: &DifferDatabase<'_>) -> bool {
```
Comment (pimeys) [inline migration-engine/connectors/sql-migration-connector/src/sql_schema_differ.rs:438]: You don't need both: the flavour and the database in here!

**Review by pimeys — APPROVED** [review]
Also noted about writing some comments in Slack. Looks good, some small fixes needed still and then ready to merge.

File: migration-engine/connectors/sql-migration-connector/src/sql_schema_differ.rs [hunk 4]
```diff
@@ -28,53 +28,21 @@ pub(crate) fn calculate_steps(schemas: Pair<&SqlSchema>, flavour: &dyn SqlFlavou
     let db = DifferDatabase::new(schemas, flavour);
     let differ = SqlSchemaDiffer { schemas, flavour, db };
     let mut steps: Vec<SqlMigrationStep> = Vec::new();
-    differ.push_create_tables(&mut steps);
 
     let tables_to_redefine = differ.flavour.tables_to_redefine(&differ);
```
[the remark this answers is not in the document]
Comment (tomhoule) [reply migration-engine/connectors/sql-migration-connector/src/sql_schema_differ.rs:32]: Yes, eventually. Next PR probably.

File: migration-engine/connectors/sql-migration-connector/src/sql_schema_differ.rs (hunk 3 above)
Comment (tomhoule) [reply migration-engine/connectors/sql-migration-connector/src/sql_schema_differ.rs:438]: Good catch!

File: migration-engine/connectors/sql-migration-connector/src/sql_schema_differ.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (tomhoule) [reply migration-engine/connectors/sql-migration-connector/src/sql_schema_differ.rs:32]: I tried doing this in a new commit — it's not complicated but a big diff, let's do it next PR.

File: migration-engine/connectors/sql-migration-connector/src/pair.rs (hunk 1 above)
Comment (tomhoule) [reply migration-engine/connectors/sql-migration-connector/src/pair.rs]: Done

**Pull request merged**